config/src/config.rs

```diff
@@ -180,6 +181,17 @@ impl NodeConfig {
         }
         for network in &mut config.full_node_networks {
             network.load(path.as_ref())?;
+            // For non-validator roles, the peer_id should be derived from the network identity key.
+            assert_eq!(
+                network.peer_id,
+                PeerId::try_from(
+                    network
+                        .network_keypairs
+                        .get_network_identity_public()
+                        .to_bytes()
+                )
+                .unwrap()
+            );
```

Comment:
This should probably not be an assert, which would crash the program here, instead return an Error, potentially using the `ensure!()` macro that is used above.

Reply:
Ahh, yes, I'll make another commit that addresses the woeful state of error handling that is pervasive in this code. In retrospect, I should have started with that one though as it is a straight forward building block....